asab/application.py

```diff
@@ -28,19 +29,56 @@
 
 
 class Application(metaclass=Singleton):
+	"""The base application object that maintains the global application state.
+
+	You can provide your own implementation by creating a subclass. It is intended to be a Singleton.
+
+	Examples:
+	```python
+	class MyApplication(asab.Application):
+		async def main(self):
+			print("Hello world!")
+			self.stop()
+
+	if __name__ == "__main__":
+		app = MyApplication()
+		app.run()
+	```
+	"""
 
 	Description = "This app is based on ASAB."
 
-	def __init__(self, args=None, modules=[]):
-		'''
-		Argument `modules` allows to specify a list of ASAB modules that will be added by `app.add_module()` call.
+	def __init__(self, args: typing.Optional[list] = None, modules: list = []):
+		"""
+		Initialize the Application provided with arguments and modules.
+
+		Args:
+			args: sequence of arguments to be parsed by `Application.parse_arguments()` call.
+			modules: list of ASAB modules to be added by `Application.add_module()` call.
 
-		Example:
+		Examples:
 
+		```python
 		class MyApplication(asab.Application):
 			def __init__(self):
 				super().__init__(modules=[asab.web.Module, asab.zookeeper.Module])
-		'''
+		```
+		"""
+
+		self.ExitCode: typing.Union[str, int]
```

Comment:
![image](https://github.com/TeskaLabs/asab/assets/68327081/a0bdfeca-2a68-4ae1-be61-5d072685f888)

Reply:
`typing.Union[int, str]` ... `int` is much more important